internal/jobobject/queue_test.go

```diff
@@ -0,0 +1,89 @@
+package jobobject
+
+import (
+	"testing"
+	"time"
+)
+
+func TestReadOrWait(t *testing.T) {
+	q := newMessageQueue()
+
+	// What we care about for job objects. Verify that we can just read when theres a value
+	// available in the queue, otherwise block if there's not a value and wait for one to become
+	// available. This tests that a client could start reading notifications late into the lifecycle
+	// of a job when there's likely a couple of messages already queued --> flush the queue of
+	// messages, and then wait for more to arrive.
+	go func() {
+		q.write(1)
+		q.write(2)
+		q.write(3)
+		time.Sleep(time.Second * 5)
+		q.write(4)
+	}()
+
+	time.Sleep(time.Second * 2)
+
+	for {
+		if msg, err := q.readOrWait(); err != nil {
+			t.Fatal(err)
+		} else {
+			// We successfully received the last message after the timeout.
+			if msg == 4 {
+				break
+			}
+		}
+	}
```

Comment:
We should probably add a context deadline in case of an infinite hang if someone breaks something.